### ISSUE TITLE
Generate Spring Autoconfig workflow failed

### ISSUE THREAD

**Issue by JoeWang1127** [GoogleCloudPlatform/spring-cloud-gcp#3043]
Generate Spring Autoconfig workflow failed: 
```
[INFO] Scanning for projects...
Error: ] Some problems were encountered while processing the POMs:
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-workstations-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-workflows-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-workflow-executions-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-websecurityscanner-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-webrisk-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-vpcaccess-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-vmmigration-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-video-transcoder-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-video-stitcher-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-video-intelligence-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-translate-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-trace-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-tpu-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-texttospeech-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-telcoautomation-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-tasks-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-talent-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-storage-transfer-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-speech-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-shell-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-servicedirectory-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-service-usage-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-service-management-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-service-control-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-securityposture-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-securitycentermanagement-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-securitycenter-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-security-private-ca-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-securesourcemanager-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-scheduler-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-retail-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-resource-settings-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-redis-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-recommender-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
Error:  Child module /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/google-cloud-recaptchaenterprise-spring-starter of /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-previews/pom.xml does not exist @ 
```

Log: https://github.com/GoogleCloudPlatform/spring-cloud-gcp/actions/runs/9848368573/job/27190170382

To reproduce the issue, run the workflow with
pr branch name: `chore/update-spring-autogen`
fork name: none

**Comment by diegomarquezp** [comment]
From the same logs:

```
2024-07-08T23:38:27.9443676Z ERROR: /home/runner/work/spring-cloud-gcp/spring-cloud-gcp/spring-cloud-generator/googleapis/google/cloud/alloydb/connectors/v1/BUILD.bazel:160:1: name 'java_gapic_spring_library' is not defined
```

Interesting 🤔, I thought I merged [this PR](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/2985/files)
I fell into your exact same problem: alloydb connector + gke-connect-gateway don't have the `java_gapic_library` rule in their BUILD.bazel file. I added them in an exclusion list.

I'll reduce the PR to only have the script changes so we can run and generate the autoconfigs from the latest protos

**Comment by diegomarquezp** [comment]
Fixed in https://github.com/googleapis/sdk-platform-java/pull/3019#discussion_r1671030813

**Comment by blakeli0** [comment]
> Fixed in [googleapis/sdk-platform-java#3019 (comment)](https://github.com/googleapis/sdk-platform-java/pull/3019#discussion_r1671030813)

The link is pointing to the DirectPath issue, I guess it is a mistake?
